system_tests/program_gas_test.go

```diff
@@ -385,6 +390,10 @@ func evmOpcodesGasUsage(ctx context.Context, rpcClient rpc.ClientInterface, tx *
 		op := vm.StringToOp(result.StructLogs[i].Op)
 		gasUsed := uint64(0)
 		if op == vm.CALL || op == vm.STATICCALL || op == vm.DELEGATECALL || op == vm.CREATE || op == vm.CREATE2 {
+			if result.StructLogs[i].GasCost == 0 {
+				// ignore mock call emitted by arbos
```

Comment:
I don't think we want to inject these logs into normal EVM execution, unless a precompile call was part of execution or similar.

Reply:
Yes, that makes sense. These happen at the end of the execution, regardless of whether the contract was an EVM contract or a stylus one. They were not being injected before geth 1.14. I wonder why it changed.

Reply:
the extra call logged comes from arbos/util/transfer.go TransferBalance: info.MockCall.
It seems like a problem in the condition (if/else) but I want to learn more about the code there.